### PR TITLE
Support showing version check error message in WebUI

### PR DESCRIPTION
**Pull request by SparkSnail**

### CODE REVIEW

File: src/nni_manager/training_service/common/clusterJobRestServer.ts [hunk 1]
```diff
@@ -41,6 +41,10 @@ export abstract class ClusterJobRestServer extends RestServer{
 
     private readonly expId: string = getExperimentId();
 
+    private openVersionCheck: boolean = true; //switch to open version check
```
Comment (yds05) [inline src/nni_manager/training_service/common/clusterJobRestServer.ts]: rename to 'enableVersionCheck'

File: src/nni_manager/training_service/common/clusterJobRestServer.ts [hunk 2]
```diff
@@ -58,6 +62,14 @@ export abstract class ClusterJobRestServer extends RestServer{
         }
         return this.port;
     }
+    
+    public get getErrorMessage(): string | undefined{
+        return this.errorMessage;
+    }
+    
+    public set setOpenVersionCheck(versionCheck: boolean) {
```
Comment (yds05) [inline src/nni_manager/training_service/common/clusterJobRestServer.ts]: Rename to setDisableVersionCheck

File: tools/nni_trial_tool/trial_keeper.py [hunk 3]
```diff
@@ -118,13 +121,20 @@ def check_version(args):
         try:
             trial_keeper_version = regular.search(trial_keeper_version).group('version')
             nni_log(LogType.Info, 'trial_keeper_version is {0}'.format(trial_keeper_version))
-            training_service_version = regular.search(args.version).group('version')
-            nni_log(LogType.Info, 'training_service_version is {0}'.format(training_service_version))
-            if trial_keeper_version != training_service_version:
+            nni_manager_version = regular.search(args.version).group('version')
```
Comment (yds05) [inline tools/nni_trial_tool/trial_keeper.py]: suggest to change args.version to args.nni_manager_version.

File: src/nni_manager/training_service/common/clusterJobRestServer.ts (hunk 1 above)
Comment (SparkSnail) [reply src/nni_manager/training_service/common/clusterJobRestServer.ts]: changed to `enableVersionCheck`

File: src/nni_manager/training_service/common/clusterJobRestServer.ts (hunk 2 above)
Comment (SparkSnail) [reply src/nni_manager/training_service/common/clusterJobRestServer.ts]: change to `setEnableVersionCheck`

File: tools/nni_trial_tool/trial_keeper.py (hunk 3 above)
Comment (SparkSnail) [reply tools/nni_trial_tool/trial_keeper.py]: changed to `nni_manager_version`

File: src/nni_manager/training_service/common/clusterJobRestServer.ts [hunk 4]
```diff
@@ -77,6 +89,26 @@ export abstract class ClusterJobRestServer extends RestServer{
...
+        router.post(`/version/${this.expId}/:trialId`, (req: Request, res: Response) => {
+            if (this.enableVersionCheck) {
+                try {
+                    if (req.body.tag === 'VCSuccess') {
+                        this.log.info(`Version check in trialKeeper success!`);
+                        this.versionCheckSuccess = true;
+                    } else {
+                        this.errorMessage = req.body.msg;
+                    }
+                } catch(err) {
```
Comment (chicm-ms) [inline src/nni_manager/training_service/common/clusterJobRestServer.ts]: Suggest handle the case that different trial jobs report inconsistent results

Comment (SparkSnail) [reply src/nni_manager/training_service/common/clusterJobRestServer.ts]: add a logic to handle inconsistent result.

File: src/nni_manager/training_service/common/clusterJobRestServer.ts [hunk 5]
```diff
@@ -77,6 +89,32 @@ export abstract class ClusterJobRestServer extends RestServer{
...
 
+        router.post(`/version/${this.expId}/:trialId`, (req: Request, res: Response) => {
+            if (this.enableVersionCheck) {
+                try {
+                    const checkResultSuccess: boolean = req.body.tag === 'VCSuccess'? true: false;
+                    if (this.versionCheckSuccess !== undefined && this.versionCheckSuccess !== checkResultSuccess) {
+                        this.errorMessage = 'Version check error, version check result is inconsistent!';
+                        this.log.error(this.errorMessage);
+                    }
+                    else if (checkResultSuccess) {
```
Comment (chicm-ms) [inline src/nni_manager/training_service/common/clusterJobRestServer.ts]: Please fix this line indent.

Comment (SparkSnail) [reply src/nni_manager/training_service/common/clusterJobRestServer.ts]: fixed.

**Pull request merged**